apps/src/blockly/addons/functionEditor.ts

```diff
@@ -580,6 +625,19 @@ export default class FunctionEditor {
     // procedure definition.
     Blockly.Events.disable();
     this.editorWorkspace.clear();
+    const primaryWorkspaceVariableMap = this.primaryWorkspace?.getVariableMap();
```

Comment:
Can you add a comment explaining why we need to do this?

Reply:
Oh yes, good call! The reason is that the previous line wipes the map, so  we rebuild it manually here. (This is also why we can’t just use the same map for all workspaces.)